src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs

```diff
@@ -312,16 +332,25 @@ class Test
 
             var spans = await TestDiagnosticTagProducer<DiagnosticsSquiggleTaggerProvider, IErrorTag>.GetErrorsFromUpdateSource(workspace, updateArgs, DiagnosticKind.CompilerSyntax);
 
-            Assert.Equal(2, spans.Count());
-            var first = spans.First();
-            var second = spans.Last();
+            if (pull)
```

Comment:
this is a case where there is an intentional deviation.  in pull diagnostics, we know we're always getting accurate diagnostics against the document, so we always use *all* of the diagnostics produced. With push diagnostics though, the diagnostics may be against a different snapshot, and that means their length might become 0 as we clamp them to the buffer.  in that case, we filter them out.  this test tests exactly that, so we get different results.